sample/core.py

```diff
@@ -161,18 +171,18 @@ def get_categoryinfo(self):
                     continue
                 if page_id not in ci_list:
                     ci_list[page_id] = []
-                ci_list[page_id] = page_content[prop]
+                ci_list[page_id] += page_content[prop]
 
         self.payload.pop('cicontinue', None)
         self.payload['prop'] = None
         return ci_list
 
-    def get_duplicatefiles(self, dflimit="max", dfdir="ascending", dflocalonly=None):
+    def get_duplicatefiles(self, dflimit="max", dfdir="ascending", dflocalonly=False):
         prop = 'duplicatefiles'
         self.payload['prop'] = prop
         self.payload['dflimit'] = dflimit
         self.payload['dfdir'] = dfdir
-        if dflocalonly is not None:
+        if dflocalonly==True:
```

Comment:
- The response for `get_categoryinfo` is correct. If there is no `categoryinfo` present then there won't be a key for that `pageid` in the response. Check this [demo response](https://en.wikipedia.org/w/api.php?action=query&titles=Albert%20Einstein%7CCategory:Foo%7CCategory:Infobox%20templates&prop=categoryinfo).

- The input for `get_duplicatefiles` is a file name, so obviously it won't return a valid `pageid`.